src/lib/data.ts

```diff
@@ -118,40 +127,58 @@ export async function get_list_dir(dir_name: string, filter_name: (name: string)
 }
 
 
+/**
+ * Get a single implementation report in JSON and convert to its internal format, 
+ * 
+ * @internal
+ */
+async function get_an_implementation_report(fname: string): Promise<ImplementationReport> {
+    // Just to make the code more readable...
+    type raw_index_pair = [string, string|boolean];
+    type internal_index_pair = [string, Score];
+    interface raw_map {[index: string]: (boolean|string)}
+    interface internal_map {[index: string]: Score}
+   
+    // the boolean|string in the JSON file is transformed into a proper Score
+    const transform = (raw_score: (boolean|string)): Score => {
+        if (typeof(raw_score) === 'boolean') {
+            return (raw_score) ? Score.PASS : Score.FAIL;
+        } else {
+            return (raw_score === "n/a")? Score.NONAPPLICABLE : Score.UNTESTED;
```

Comment:
Maybe raw_score.toLowerCase() === "n/a" ? Unless that happens somewhere else and I am just missing it.

Reply:
No, it is not handled elsewhere and it is indeed a good idea. Added.

Reply:
Typescript requires me an extra "else if (raw_score === null)" to do so, though, because technically the toLowerCase() does not work on null. And typescript is absolutely right!